Podcast/Cache.swift

```diff
@@ -0,0 +1,107 @@
+            episode.seriesTitle = episodeJson["series"]["title"].stringValue
+            episode.seriesID = episodeJson["series"]["id"].stringValue
+            episode.duration = episodeJson["duration"].stringValue
+            episode.tags = episodeJson["tags"].stringValue.components(separatedBy: ";").map({ tag in Tag(name: tag) })
+            episode.audioURL = URL(string: episodeJson["audio_url"].stringValue)
+            episode.dateCreated = DateFormatter.parsingDateFormatter.date(from: episodeJson["pub_date"].stringValue) ?? Date()
+            episode.smallArtworkImageURL = URL(string: episodeJson["series"]["image_url_sm"].stringValue)
+            episode.largeArtworkImageURL = URL(string: episodeJson["series"]["image_url_lg"].stringValue)
+            return episode
+        } else {
+            let episode = Episode(json: episodeJson)
+            episodeCache[id] = episode
+            return episode
+        }
+    }
+    
+    func update(seriesJson: JSON) -> Series {
+        let id = seriesJson["id"].stringValue
+        if let series = seriesCache[id] {
+            series.title = seriesJson["title"].stringValue
+            series.smallArtworkImageURL = URL(string: seriesJson["image_url_sm"].stringValue)
+            series.largeArtworkImageURL = URL(string: seriesJson["image_url_lg"].stringValue)
+            let lastUpdatedString = seriesJson["last_updated"].stringValue
+            series.lastUpdated = DateFormatter.parsingDateFormatter.date(from: lastUpdatedString) ?? Date()
+            series.author = seriesJson["author"].stringValue
+            series.isSubscribed = seriesJson["is_subscribed"].boolValue
+            series.numberOfSubscribers = seriesJson["subscribers_count"].intValue
+            series.tags = seriesJson["genres"].stringValue.components(separatedBy: ";").map({ tag in Tag(name: tag)})
+            return series
+        } else {
+            let series = Series(json: seriesJson)
+            seriesCache[id] = series
+            return series
+        }
+    }
+    
+    func update(userJson: JSON) -> User {
+        let id = userJson["id"].stringValue
+        print(id)
+        if let user = userCache[id] {
+            user.firstName = userJson["first_name"].stringValue
+            user.lastName = userJson["last_name"].stringValue
+            user.username = userJson["username"].stringValue
+            user.numberOfFollowers = userJson["followers_count"].intValue
+            user.numberOfFollowing = userJson["followings_count"].intValue
+            user.isFollowing = userJson["is_following"].boolValue
+            user.imageURL = URL(string: userJson["image_url"].stringValue)
+            return user
+        } else {
+            let user = User(json: userJson)
+            userCache[id] = user
+            return user
+        }
+    }
+    
+    // Use this to load an episode if needed (YOU SHOULDN'T HAVE TO)
+    func load(episode id: String) -> Episode? {
```

Comment:
this function doesn't make sense to me .. if its just a wrapper on getting an item from the cache we should call it `getEpisode(id: String)` or something like that and have one for seriesCache and usersCache as well

Reply:
Yeah we shouldn't need this function. I forgot to make the others, I'll change it from load to get.